### ISSUE TITLE
Regression since 2.1.1

### ISSUE THREAD

**Issue by novemberborn** [micromatch/picomatch#59]
👋 thanks for your hard work on this package.

I have patterns like this:

```
'**/__tests__/**/__{helper|fixture}?(s)__/**/*',
'**/test?(s)/**/{helper|fixture}?(s)/**/*'
```

With the latest release these no longer match.

I'm trying to wrap up a release so don't have the time to make an isolated reproduction, but if you check out [this PR](https://github.com/avajs/ava/pull/2341) and revert the last commit, and then run `npx tap --no-cov test/globs.js` you should see the test failures.

Let me know if you'd like more details and I'll try and find the time to help out. Thanks!

**Comment by jonschlinkert** [comment]
Very sorry for the regression and thanks for creating the issue!

> check out [this PR](https://github.com/avajs/ava/pull/2341) and revert the last commit, and then run `npx tap --no-cov test/globs.js`

Can you provide a reduced test case please? I don't use the same tooling, and I have limited time for the next couple of days. If you can show me exactly what's failing I can probably push up a patch quickly.

**Comment by novemberborn** [comment]
> I have limited time for the next couple of days.

I'm in the same boat unfortunately. Will respond back when I can.

**Comment by jonschlinkert** [comment]
@novemberborn I completely missed the part of your comment where you said "don't have the time to make an isolated reproduction". Sorry about that, I think I read the comment three times before I noticed that part for some reason. I think my eyes are trained (in a bad way) to look for code lol. 

I'm looking into it, thanks.

**Comment by jonschlinkert** [comment]
@novemberborn ok I figured out the issue. In brace patterns you are using pipes instead of commas to separate patterns. 

https://github.com/avajs/ava/blob/9ea96086f85d9cb10e3b1a8cf96047615c41b855/lib/globs.js#L56-L67

For now, you can fix those tests by simply changing `|` to `,` in the brace patterns. Although this is correct, it worked before my recent patches, so It seems that I inadvertently introduced a regression. 

I will also push up a patch to allow the old behavior to work... specifically, braces with a single pattern are regarded as literals, and since `{a|b}` is technically a single pattern, it's converted to a literal.

**Comment by novemberborn** [comment]
@jonschlinkert thanks for looking into this!

Yes `,` makes more sense here. Not sure why I wrote it with `|`.

> I will also push up a patch to allow the old behavior to work... specifically, braces with a single pattern are regarded as literals, and since `{a|b}` is technically a single pattern, it's converted to a literal.

Should that old behavior have worked at all?

**Comment by jonschlinkert** [comment]
> Should that old behavior have worked at all?

No, probably not, but since it did it's safer to keep support for that behavior until we publish the next major release.
